build-plugins/build-support/src/main/kotlin/aws/sdk/kotlin/gradle/plugins/artifactmetrics/AnalyzeMetricsTask.kt

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package aws.sdk.kotlin.gradle.plugins.artifactmetrics
+
+import aws.sdk.kotlin.services.s3.S3Client
+import aws.sdk.kotlin.services.s3.model.GetObjectRequest
+import aws.smithy.kotlin.runtime.content.decodeToString
+import aws.smithy.kotlin.runtime.io.use
+import kotlinx.coroutines.runBlocking
+import org.gradle.api.DefaultTask
+import org.gradle.api.GradleException
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.TaskAction
+import java.io.File
+import kotlin.math.abs
+
+internal abstract class AnalyzeMetricsTask : DefaultTask() {
+    @get:InputFile
+    abstract val metricsFile: RegularFileProperty
+
+    @get:OutputFile
+    abstract val summary: RegularFileProperty
+
+    @get:OutputFile
+    abstract val shouldFailWorkflow: RegularFileProperty
+
+    init {
+        metricsFile.convention(project.layout.buildDirectory.file(OUTPUT_PATH + "artifactSizeMetrics.csv"))
+        summary.convention(project.layout.buildDirectory.file(OUTPUT_PATH + "artifact-summary.md"))
+        shouldFailWorkflow.convention(project.layout.buildDirectory.file(OUTPUT_PATH + "shouldFailWorkflow.txt"))
+    }
+
+    @TaskAction
+    fun analyze() {
+        val releaseMetricsFile =
+            project.layout.buildDirectory.file(OUTPUT_PATH + "latestReleaseMetrics.csv").get().asFile
+        writeReleaseMetrics(releaseMetricsFile)
+
+        val releaseMetrics = releaseMetricsFile.toMap()
+        val currentMetrics = metricsFile.get().asFile.toMap()
+        val artifactNames = getArtifactNames(releaseMetricsFile)
+        val analysis = analyzeMetrics(artifactNames, releaseMetrics, currentMetrics)
+
+        shouldFailWorkflow.get().asFile.writeText(analysis.significantChange.toString())
+        val diffTable = createDiffTable(analysis)
+        val output = if (analysis.delta) diffTable else noDiffMessage
+
+        summary.get().asFile.writeText(output)
+    }
+
+    private fun writeReleaseMetrics(releaseMetricsFile: File) = runBlocking {
+        S3Client.fromEnvironment().use { s3 ->
+            s3.getObject(
+                GetObjectRequest {
+                    bucket = "TODO"
```

Comment:
This is to _get_ the release metrics, the metrics are stored locally in the `build/reports/metrics` directory (`OUTPUT_PATH`). The upload to S3 will be done by the workflows though I think it would also make sense to do it in the plugin

Reply:
Oh right, this is fine then